### PR TITLE
Update example and docs based on feedback

### PR DESCRIPTION
**Pull request by geky**
Update example and docs based on feedback.

WIP, more to be added, this is in response to https://github.com/ARMmbed/mbed-os-example-filesystem/issues/9
cc @MarceloSalazar, @AnotherButler

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -172,6 +172,10 @@ Mbed OS has two options for the file system:
   The little file system (LittleFS) is a fail-safe file system we designed
   for embedded systems, specifically for microcontrollers that use flash
   storage.
+  
+  ``` cpp
+  LittleFileSystem fs("fs");
+  ```
```
Comment (geky) [inline README.md]: Added some code snippets to help users who are interchanging the different technology layers.

@AnotherButler, how do you want these bullet points formatted?

Comment (AnotherButler) [reply README.md]: This is fine.

File: README.md [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
 
 ## Hardware requirements
 
 This example uses a block device as storage. This can be either an external
-block device (one of SPI flash, DataFlash, or an SD card) or simulated on a
+block device (one of SPI flash, DataFlash or an SD card) or simulated on a
 heap block device on boards with enough RAM.
 
-By default, this example uses an instance of LittleFS on external SPI flash.
+This example uses an instance of the LittleFileSystem API (LittleFS) on external SPI flash.
```
Comment (geky) [inline README.md]: @AnotherButler, why no by default? Do we not want to emphasize out different filesystem/block devices are supported?

Comment (AnotherButler) [reply README.md]: First, it's unnecessary information making the sentence longer than it needs to be. Second, it's confusing to say an example uses something by default when we mean the example uses something, but you, the user, can change that something.

Comment (geky) [reply README.md]: Okeedokee, I can't think of better wording.

**Comment by MarceloSalazar** [comment]
This is now looking really good! Thanks @geky!

**Comment by geky** [comment]
@adbridge, Thanks!

**Pull request merged**